### PR TITLE
FitsToJson.cpp with gtests

### PR DESCRIPTION
**Pull request by amystamile-usgs**
## Description

Added FitstoJson object to handle json conversion to prepare for fits label ingestion in isisimport. Fits label to pvl logic pulled from ProcessImportFits.

## Motivation and Context

PDS4 Sprint

## How Has This Been Tested?

Added tests in this PR.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/FitsToJson/FitsToJson.cpp [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+namespace Isis {
+
+  /**
+   * Convert the contents of a stream to a JSON object.
+   *
+   * @param the stream to convert
+   * @return @b json The contents of the stream as a JSON object
+   */
+  json fitsToJson(std::ifstream &fileStream) {
```
Comment (acpaquette) [inline isis/src/base/objs/FitsToJson/FitsToJson.cpp:30]: Could we add a small comment in the description that this logic was taken from processimportfits?

**Review by acpaquette — COMMENTED** [review]
Overall looks good! One comment to address and just waiting for CI to come back

File: isis/src/base/objs/FitsToJson/FitsToJson.cpp [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+      }
+    }
+    return pvlObjectToJSON(fitsPvl);
+  }
+
+
+  /**
+   * Converts a FITS file to a json object.
+   *
+   * @param fitsFile Path to an FITS file.
```
Comment (scsides) [inline isis/src/base/objs/FitsToJson/FitsToJson.cpp]: In this case FITS is pronounced, so the "an" should be "a"

**Review by scsides — DISMISSED** [review]
One silly comment, looks good

**Review by acpaquette — APPROVED** [review]

**Pull request merged**